StripePaymentSheet/StripePaymentSheet/Source/Helpers/STPClientAttributionMetadata+PaymentSheet.swift

```diff
@@ -6,12 +6,30 @@
 //
 
 @_spi(STP) import StripeCore
+@_spi(STP) import StripePayments
 
 extension STPClientAttributionMetadata {
     static func makeClientAttributionMetadataIfNecessary(analyticsHelper: PaymentSheetAnalyticsHelper, intent: Intent, elementsSession: STPElementsSession) -> STPClientAttributionMetadata? {
         if analyticsHelper.integrationShape.isMPE {
-            return intent.clientAttributionMetadata(elementsSessionConfigId: elementsSession.sessionID)
+            return makeClientAttributionMetadata(intent: intent, elementsSession: elementsSession)
         }
         return nil
     }
+
+    static func makeClientAttributionMetadata(intent: Intent, elementsSession: STPElementsSession) -> STPClientAttributionMetadata {
+        switch intent {
+        case .paymentIntent(let paymentIntent):
+            return .init(elementsSessionConfigId: elementsSession.sessionID,
+                         paymentIntentCreationFlow: .standard,
+                         paymentMethodSelectionFlow: paymentIntent.automaticPaymentMethods?.enabled ?? false ? .automatic : .merchantSpecified)
+        case .setupIntent(let setupIntent):
+            return .init(elementsSessionConfigId: elementsSession.sessionID,
+                         paymentIntentCreationFlow: .standard,
+                         paymentMethodSelectionFlow: setupIntent.automaticPaymentMethods?.enabled ?? false ? .automatic : .merchantSpecified)
+        case .deferredIntent(let intentConfig):
+            return .init(elementsSessionConfigId: elementsSession.sessionID,
+                         paymentIntentCreationFlow: .deferred,
+                         paymentMethodSelectionFlow: intentConfig.paymentMethodTypes?.isEmpty ?? true ? .automatic : .merchantSpecified)
```

Comment:
Why does this one default to true but the other two don't? Is this intentional?

Reply:
SGTM, can we add a comment to these clarifying? I think it could appear to be a mistake/bug to others without knowing why.